shell/src/main/java/tachyon/shell/TFsShell.java

```diff
@@ -166,73 +152,66 @@ private int loadPath(TachyonFS tachyonClient, TachyonURI filePath) throws IOExce
+    System.out.println("Copied " + srcPath + " to " + dstPath);
   }
 
-  private int copyPath(File src, TachyonFS tachyonClient, TachyonURI dstPath) throws IOException {
+  private void copyPath(File src, TachyonFS tachyonClient, TachyonURI dstPath) throws IOException {
     if (!src.isDirectory()) {
       TachyonFile tFile = tachyonClient.getFile(dstPath);
       if (tFile != null && tFile.isDirectory()) {
         dstPath = dstPath.join(src.getName());
       }
       int fileId = tachyonClient.createFile(dstPath);
-      if (fileId == -1) {
-        return -1;
-      }
-      tFile = tachyonClient.getFile(fileId);
-      Closer closer = Closer.create();
       try {
-        WriteType writeType =
-            mTachyonConf.getEnum(Constants.USER_DEFAULT_WRITE_TYPE, WriteType.CACHE_THROUGH);
-        OutStream os = closer.register(tFile.getOutStream(writeType));
-        FileInputStream in = closer.register(new FileInputStream(src));
-        FileChannel channel = closer.register(in.getChannel());
-        ByteBuffer buf = ByteBuffer.allocate(8 * Constants.MB);
-        while (channel.read(buf) != -1) {
-          buf.flip();
-          os.write(buf.array(), 0, buf.limit());
+        tFile = tachyonClient.getFile(fileId);
+        Closer closer = Closer.create();
+        try {
+          OutStream os =
+                  closer.register(tFile.getOutStream(mTachyonConf.getEnum(
+                          Constants.USER_DEFAULT_WRITE_TYPE, WriteType.CACHE_THROUGH)));
+          FileInputStream in = closer.register(new FileInputStream(src));
+          FileChannel channel = closer.register(in.getChannel());
+          ByteBuffer buf = ByteBuffer.allocate(8 * Constants.MB);
+          while (channel.read(buf) != -1) {
+            buf.flip();
+            os.write(buf.array(), 0, buf.limit());
+          }
+        } finally {
+          closer.close();
         }
-      } finally {
-        closer.close();
+      } catch (IOException e) {
+        // Try to delete the file we failed to copy, so that we don't have an empty file lying around
+        tachyonClient.delete(fileId, false);
+        throw e;
       }
-      return 0;
     } else {
       tachyonClient.mkdir(dstPath);
       for (String file : src.list()) {
         TachyonURI newPath = new TachyonURI(dstPath, new TachyonURI(file));
         File srcFile = new File(src, file);
-        if (copyPath(srcFile, tachyonClient, newPath) == -1) {
-          return -1;
-        }
+        copyPath(srcFile, tachyonClient, newPath);
       }
     }
-    return 0;
   }
 
   /**
    * Copies a file specified by argv from the filesystem to the local filesystem.
    *
    * @param argv [] Array of arguments given by the user's input from the terminal
-   * @return 0 if command is successful, -1 if an error occurred.
    * @throws IOException
    */
-  public int copyToLocal(String[] argv) throws IOException {
+  public void copyToLocal(String[] argv) throws IOException {
```

Comment:
We can first write to a local temporary file, if the write is successful, then we rename the temporary file to the destination file, otherwise, just delete the temporary file and leave the destination file unmodified(if it exists before).